test/tests/server_connectorTest.js

```diff
@@ -14,10 +14,12 @@ describe("Connector Server", function () {
 		
 		win = yield loadZoteroPane();
 		connectorServerPath = 'http://127.0.0.1:' + Zotero.Prefs.get('httpServer.port');
-		testServerPath = 'http://127.0.0.1:' + testServerPort;
 	});
 	
 	beforeEach(function () {
+		// Alternate ports to prevent exceptions not catchable in JS
+		testServerPort += (testServerPort & 1) ? 1 : -1;
```

Comment:
Is this actually reliable? That is, when the port is left open, is it eventually closed, such that reusing it later might work, or does it just remain open, such that this would only work if it happens once? (And wouldn't this be subject to a race condition if the port remained opened long enough for this to be run twice?)